cat-launcher/src/pages/SettingsPage/lib/utils.ts

```diff
@@ -0,0 +1,27 @@
+import { ThemeColors } from "@/generated-types/ThemeColors";
+
+export function isThemeEqual(
+  themeColors: ThemeColors,
+  settingsColors: ThemeColors,
+): boolean {
+  const tcRecord = themeColors as unknown as Record<
```

Comment:
The function `isThemeEqual` uses `as unknown as Record<...>` which bypasses TypeScript's type safety. A more type-safe implementation would be to iterate over the keys of the `ThemeColors` object, which would be more robust and less prone to runtime errors if the `ThemeColors` type changes.



```
  const keys = Object.keys(themeColors) as (keyof ThemeColors)[];
  if (keys.length !== Object.keys(settingsColors).length) {
    return false;
  }

  return keys.every((key) => {
    const tc = themeColors[key];
    const sc = settingsColors[key];

    if (!tc || !sc) {
      return tc === sc;
    }

    return tc[0] === sc[0] && tc[1] === sc[1] && tc[2] === sc[2];
  });
}
```